### PR TITLE
Feature/complete octree3

### LINKED ISSUE #8: Algo 3
write algo3

### PR DESCRIPTION
**Pull request by mazzeaar**
This is an implementation of algorithm 3.
I left out the first 3 lines of the algo, as those are not implemented yet.
I would suggest to leave those out of algo3 and perform the operations outside the function, this should make testing easier later on.

### CODE REVIEW

**Comment by mazzeaar** [comment]
DO NOT MERGE YET, I DID AN OOPISE

**Comment by mazzeaar** [comment]
oopsie fixed, is mergeable

**Review by gfreinersimon — APPROVED** [review]
looks fine, needs to be tested after merging lol

**Pull request merged**
